spec/requests/sessions_spec.rb

```diff
@@ -0,0 +1,152 @@
+# frozen_string_literal: true
+
+describe 'Sessions', type: :request do
+  let(:recaptcha_id) { 'recaptcha_form' }
+
+  describe '#new' do
+    after { cookies[:login_recaptcha] = nil }
+
+    context 'when recaptcha is not configured' do
+      before do
+        allow(RecaptchaService).to receive(:recaptcha_configured?).and_return(false)
+      end
+
+      it 'does not display recaptcha' do
+        cookies[:login_recaptcha] = true
+        get new_user_session_path
+
+        expect(response.body).to_not include recaptcha_id
+      end
+    end 
+
+    context 'when recaptcha is configured' do
+      before do
+        allow(RecaptchaService).to receive(:recaptcha_configured?).and_return(true)
+      end
+
+      context 'and user has recaptcha cookie' do
```

Comment:
```suggestion
      context 'when user has recaptcha cookie' do
```

Let's change these, line 36-37, 51-52, 60-61, 71, 112, 124, 136, 142 to say `when` instead. For those 2 nested context, let's just merge to 2 ands into one lines as well since we don't need to distinguish the other cases since we don't have other `it` blocks to separate the 2 cases. 

See https://www.betterspecs.org/#contexts for RSpec best practices.